src/backend/base/langflow/custom/custom_component/component.py

```diff
@@ -786,7 +787,11 @@ def _set_input_value(self, name: str, value: Any) -> None:
 
     def _validate_outputs(self) -> None:
         # Raise Error if some rule isn't met
-        pass
+        if self.selected_output is not None:
+            output_names = [output.name for output in self.outputs]
+            if self.selected_output not in output_names:
+                msg = f"selected_output '{self.selected_output}' is not valid. Must be one of: {output_names}"
+                raise ValueError(msg)
```

Comment:
You can get the names from `self._outputs_map`
```suggestion
            if self.selected_output not in self._outputs_map:
                output_names = ', '.join(list(self._outputs_map.keys()))
                msg = f"selected_output '{self.selected_output}' is not valid. Must be one of: {output_names}"
                raise ValueError(msg)
```